docs/process/ci_build.md

```diff
@@ -0,0 +1,89 @@
+  * The `on` key config shows that the workflow is triggered only when a commit is made on `master`. Since the repository only allows commits to `master` through the Pull Request process, this workflow is unhelpful for Pull Request authors and reviewers because pull requests originating from any branch not called `master` (all of them) will not trigger the workflow.
+  * There is one job, which is to build and run unit tests
+  * The first step of any job is usually to checkout the latest code in the repository. Otherwise, the fresh VM runner has an empty filesystem.
+  * The first step of this job invokes `actions/checkout@v2`. This means to invoke the Github Action kept in the repository https://www.github.com/actions/checkout, and use the state of that repository that the `v2` ref points to. Since `v2` is a git "ref", that means could be a git tag, a git branch name, or a git commit hash value.
+    * The preferred convention by Github Actions is that action repository maintainers use `v1` and `v2` as tags that point to the commits that are tagged with the latest version within that major version. (Ex: `v1` points to `v1.2.3`, `v2` points to `v2.8.9.1`). Not all actions adhere to this guideline, and instead only have `vX.Y.Z` tags without a `vX` tag to the latest `vX.*.*`.
+  * In this simplistic example, the way to allow a Pull Request to run the same checks after each new commit is pushed to the PR, in the same way that checks are run after a merge to `master`, is to change `.on.pull_request.branches` to [match all origin branch names](https://github.com/unicode-org/icu4x/commit/16ae4611738fbe94b36e17b77aee6cc541c0a171).
+
+## Slightly Advanced Features Used in unicode-org repos
+
+* A "job matrix" allows you to run a collection of parameterized jobs, where you indicate which parts of the job configuration are the parameters (variables) and what values they are allowed to take. If the parameter is the OS of the VM, then you can run the same job on Linux, macOS, and Windows with little extra work
+  * The parameters are defined as fields under the `strategy.matrix` key within the job, and the range of allowed values are stored as arrays. The parameters are used (string interpolated) with `${{ matrix.<param> }}` syntax. [Example](https://github.com/unicode-org/icu4x/blob/master/.github/workflows/build-test.yml#L17-L21):
+```yml
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - ...
+```
+  * A job matrix can help decrease wall clock time for multiple independent long-running steps, like benchmarks. [Example](https://github.com/unicode-org/icu4x/blob/master/.github/workflows/build-test.yml#L115-L127):
+```
+  benchmark:
+    strategy:
+      fail-fast: false
+      matrix:
+        component:
```

Comment:
'component:' looks interesting but is introduced out of the blue. Please add a brief description how components are defined.

Reply:
Done. I added a small description.

Btw: `component` is merely another example of a parameter in a parameterized job, as configured by the job matrix.